microservices/sfts/README.md

```diff
@@ -96,19 +98,23 @@ The structure of the config file should resemble the following:
   "destination": String,
   "object_prefix": String,
   "dml": String,
-  "sfts_path": String
+  "header": Boolean,
+  "sfts_path": String,
+  "extension": String
 }
 ```
 
-The keys in the config file are defined as:
+The keys in the config file are defined as follows. All parameters are required in order to use one configuration file for both scripts (which is recommended for service encapsulation and ease of maintenance):
 
 - `"bucket"`: the label defining the S3 bucket that the microservice will reference.
-- `"source"`: the first prefix of source objects, as in: `s3://<bucket>/<source>/.../<object>`.
-- `"directory"`: the last path prefix before the object itself: `s3://<bucket>/<source>/<directory>/<object>` or `s3://<bucket>/<destination>/<good|bad|batch>/<source>/<directory>/<object>`
-- `"destination"`: the first prefix of processed objects, as in `s3://<bucket>/<destination>/<good|bad|batch>`.
+- `"source"`: the first prefix of source objects, as in: `"s3://<bucket>/<source>/.../<object>"`.
+- `"directory"`: the last path prefix before the object itself: `"s3://<bucket>/<source>/<directory>/<object>"` or `"s3://<bucket>/<destination>/<good|bad|batch>/<source>/<directory>/<object>"`
+- `"destination"`: the first prefix of processed objects, as in `"s3://<bucket>/<destination>/<good|bad|batch>"`.
 - `"object_prefix"`: The final prefix of the object; treat this as a prefix on the filename itself.
-- `"dml"`: The filename under the `./dml`(./dml/) directory in this repository that contains the SQL statement to run as an UNLOAD command.
+- `"dml"`: The filename under the [`./dml`](./dml/) directory in this repository that contains the SQL statement to run as an UNLOAD command.
+- `"header"`: Setting this to true will write a first row of column header values; setting as false will omit that row.
 - `"sfts_path"`: The folder path in SFTS where the objects retrieved from S3 will be uploaded to.
+- `"extension"`: A postfix to the file name. As an extension, it must include the "`.`" character before the extension type, such as: `".csv"`. If no extension is needed then the value should be an empty string, like `""`.
```

Comment:
If extension is only intended to be used by s3_to_sfts.py, I think this rename-during-transfer behavior as you explained to me should be documented here because it is by no means a common practice. What's intuitive to me is to use the extension during file generation (i.e. in redshift_to_s3.py) and preserve the extension throughout the transfer.

Reply:
@f-w 

I used `download_file()`([documentation](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3.html#S3.Client.download_file)) from Boto3. It requires explicitly passing the object key and the filename as distinct values.

My intuition was that the simplest change to accomplish the intended outcome was to modify the filename passed to that argument (to append the _extension_ string).

The alternative is what you suggest: modifying `redshift_to_s3.py`. This would require quite a bit more complexity with several additional Boto3 function calls (copy and delete, since there is no such direct key renaming function, [as explained in this stackoverflow post](https://stackoverflow.com/questions/34492462/how-to-rename-objects-boto3-s3)) on the UNLOAD objects that had been created in S3.

I will further adjust the README file to explain that the extension is appended not at the s3 object level, but only at the file level immediately prior to transfer to SFTS.